src/test/java/io/airlift/compress/zstd/UtilTest.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.airlift.compress.zstd;
+
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import static io.airlift.compress.zstd.Util.get24BitLittleEndian;
+import static io.airlift.compress.zstd.Util.put24BitLittleEndian;
+import static org.testng.Assert.assertEquals;
+import static sun.misc.Unsafe.ARRAY_BYTE_BASE_OFFSET;
+
+public class UtilTest
+{
+    @DataProvider(name = "test24bitIntegers")
+    public static Object[][] test24bitIntegers()
+    {
+        return new Object[][] {
+                {new byte[]{1, 0, 0, 0}, 1},
+                {new byte[]{12, -83, 0, 0}, 44300},
+                {new byte[]{0, 0, -128}, 8388608},
+                {new byte[]{(byte) 0xFF, (byte) 0xFF, (byte) 0xFF}, 16777215},
+                {new byte[]{63, 25, 72, 0}, 4725055}
+        };
+    }
+
+    @Test(dataProvider = "test24bitIntegers")
+    public void test24bitIntegers(byte[] bytes, int value)
```

Comment:
Add tests for reading and writing at non-0 offset within the array.

Also, please addition of tests for `put24BitLittleEndian` in a separate commit in this PR, since they are unrelated to the bug being fixed here.

Reply:
Added non-0 offset test.

Moved `put24BitLittleEndian` to their own test and in their own commit.